tools/tokenizer/src/Tokenizer.java

```diff
@@ -605,18 +608,20 @@ private static void mappingBackText(String rawFileName, String targetFileName, S
             BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(targetFileName), "UTF8"));
 
             String buffer = "";
-            boolean isPhrase = false;
+            Double isPhrase = NOT_PHRASE;

```

Comment:
boolean?

Reply:
I changed it to Double so the program can use the same variable for keeping the scores and checking if a phrase is active.

`NOT_PHRASE` is equivalent to `isPhrase=false`. And now, cases in which `isPhrase=true` are also keeping the phrase quality score.

Reply:
in this case, let's rename it as isPhrase as phraseScore?

Reply:
Thats a better name. Will do.